src/sentry/integrations/issues.py

```diff
@@ -110,14 +122,44 @@ def store_issue_last_defaults(self, project_id, data):
               differentiation is made between the two field configs.
         """
         persisted_fields = self.get_persisted_default_config_fields()
-        if not persisted_fields:
-            return
+        if persisted_fields:
+            project_defaults = {k: v for k, v in six.iteritems(data) if k in persisted_fields}
+            self.org_integration.config.setdefault("project_issue_defaults", {}).setdefault(
+                six.text_type(project.id), {}
+            ).update(project_defaults)
+            self.org_integration.save()
+
+        user_persisted_fields = self.get_persisted_user_default_config_fields()
+        if user_persisted_fields:
+            user_defaults = {}
+            defaults_user_option_key = self._get_defaults_user_option_key()
+            user_defaults.update(
+                UserOption.objects.get_value(
+                    user=user, key=defaults_user_option_key, default={}, project=project
+                )
+            )
+            user_defaults.update(
+                {k: v for k, v in six.iteritems(data) if k in user_persisted_fields}
+            )
+            UserOption.objects.set_value(
+                user=user, key=defaults_user_option_key, value=user_defaults, project=project
+            )
+
+    def get_defaults(self, project, user):
+        project_defaults = self.get_project_defaults(project.id)
+
+        defaults_user_option_key = self._get_defaults_user_option_key()
+        user_defaults = UserOption.objects.get_value(
+            user=user, key=defaults_user_option_key, default={}, project=project
+        )
 
-        defaults = {k: v for k, v in six.iteritems(data) if k in persisted_fields}
+        defaults = {}
+        defaults.update(project_defaults)
+        defaults.update(user_defaults)
 
-        self.org_integration.config.update({"project_issue_defaults": {project_id: defaults}})
-        self.org_integration.save()
+        return defaults
 
+    # TODO(saif): Make private and move all usages over to `get_defaults`
```

Comment:
Just noting that making this private and inlining it into `get_defaults` means that a bunch of integration that don't yet have any user-scoped defaults will have to start taking in user when they didn't previously.

I think this makes sense to avoid the following confusion: integration developers have to make the cognizant decision between get_defaults and get_project_defaults, where if they choose get_project_defaults, then `get_persisted_user_default_config_fields ` will seemingly not work as expected.